internal/terraform/node_resource_plan_instance.go

```diff
@@ -547,6 +554,88 @@ func (n *NodePlannableResourceInstance) managedResourceExecute(ctx EvalContext)
 	return diags
 }
 
+// planActionTriggers plans all actions (potentially) triggered by this
+// resource instance change sequentially, and returns any diagnostics
+func (n *NodePlannableResourceInstance) planActionTriggers(ctx EvalContext, change *plans.ResourceInstanceChange) tfdiags.Diagnostics {
+	var diags tfdiags.Diagnostics
+
+	// TODO: When is the config nil?
```

Comment:
// TODO: When is the config nil?
A huge percentage of checks like that inside terrafrom are just for tests, so we get errors instead of panics when setting things up wrong.